packages/react/src/components/Icon/index.tsx

```diff
@@ -8,24 +8,45 @@ export interface IconProps extends React.HTMLAttributes<HTMLDivElement> {
   type: string;
```

Comment:
I know it's a pain, but I _strongly_ suggest making `type` stricter here.

Something like:

```ts
type: 'Thing' | 'OtherThing' | 'etc...'
```

Reply:
I don't disagree, but we would need to make sure the type stay in sync with icons.

Reply:
Actually, typescript seems to be unhappy with dynamic strings when we define types.

```sh
src/index.ts → lib...
[!] (plugin typescript) Error: @rollup/plugin-typescript TS2322: Type 'string' is not assignable to type '"menu" | "code" | "arrow-up" | "arrow-down" | "arrow-left" | "arrow-right" | "bolt" | "checkbox-checked" | "checkbox-unchecked" | "chevron-up" | "chevron-down" | "chevron-left" | ... 21 more ... | "trash"'.
src/components/ExpandCollapsePanel/PanelTrigger.tsx (29:13)
29       <Icon type={`chevron-${open ? 'down' : 'right'}`} />
```

That's solvable by using `${open ? 'chevron-down' : 'chevron-right'}`, but I think for now I'd like to leave this as a non-breaking change.

Reply:
Ah bummer, OK. Thanks for trying this out.